### DISCUSSION TITLE
Geomap: updating query based on map bounds

### DISCUSSION THREAD

**Discussion by rudibroekhuizen** [grafana/grafana#67074] [Dashboards, Panels & Visualizations]
It would be nice to be able to narrow down the search area by using map bounds, like in the [trackmap panel](https://github.com/alexandrainst/alexandra-trackmap-panel) developed by the Alexandra Institute. With this feature it is possible to build a more dynamic experience when zooming in or out on a the map, plotting markers more and more accurate on the map. 

To update the query dynamically based on the map bounds turn on Use map bounds in query. To use this you must manually add four variables to the dashboard (via settings in the top right corner). Add four variables of type constant with names minLat, minLon, maxLat, and maxLon. The values can be anything, e.g. 1, 2, 3, 4 - they will be overwritten by the plugin. The variables can then be used in a query. Here is an example that limits the query to the bounds of the map using PostgreSQL:

```
SELECT
  avg("lat") as "latitude",
  avg("long") as "longitude",
  max(abs("rssi")) as "intensity",
  max("rssi") as "tooltip",
FROM (
  SELECT sys_time, lat, long, geo, rssi, ST_ClusterKMeans(geo, 15) over() as clusters from table_name
  Where lat >= $minLat
    AND lat <= $maxLat
    AND long >= $minLon
    AND long <= $maxLon
) table_name_clustered
GROUP BY clusters
```
Example, with Clickhouse (using the h3 geospatial indexing system) as data source:

[trackmap.webm](https://user-images.githubusercontent.com/4686274/233743107-2e9a1d47-f022-4f8a-bb9e-3ada8040ea49.webm)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82722. Feel free to continue the discussion around this there. Thank you!
